src/vs/workbench/parts/debug/electron-browser/debugService.ts

```diff
@@ -240,11 +240,16 @@ export class DebugService extends ee.EventEmitter implements debug.IDebugService
 			const threadId = event.body.threadId;
 
 			this.getThreadData(threadId).then(() => {
-				this.session.stackTrace({ threadId: threadId, levels: 20 }).done((result) => {
+				let thread = this.model.getThreads()[threadId];
 
-					this.model.rawUpdate({ threadId: threadId, callStack: result.body.stackFrames, stoppedDetails: event.body });
+				this.model.rawUpdate({
+					threadId: threadId,
+					stoppedDetails: event.body,
+					allThreadsStopped: event.body.allThreadsStopped
+				});
+
```

Comment:
Looks good - I like that the changes here are minimal now.

Though I think you can handle this promises a bit better - you can nicely have only one place where you handle errors. I propose:
* Line 242: needs a `done`
* Line 251: needs a `return` and a `then`
* Line 256: needs a `return` and no `done` - this will be boobled up
* Move error handling from line 260 to 261

This way I believe the error handling and `promise` flow is nicer.